src/lib/storage/index/partial_hash/partial_hash_index_impl.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <utility>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "storage/chunk.hpp"
+#include "storage/index/abstract_table_index.hpp"
+#include "tsl/robin_map.h"
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractSegment;
+class PartialHashIndexTest;
+
+/**
+ * Base class that holds a PartialHashIndexImpl object with the correct unboxed datatype.
+ */
+class BasePartialHashIndexImpl : public Noncopyable {
+  friend PartialHashIndexTest;
+
+ public:
+  using Iterator = AbstractTableIndex::Iterator;
+  using IteratorPair = std::pair<AbstractTableIndex::Iterator, AbstractTableIndex::Iterator>;
+
+  virtual ~BasePartialHashIndexImpl() = default;
+
+  virtual size_t add(const std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>&, const ColumnID) { return 0; }
```

Comment:
please add a comment explaining the high-level functionality, including the parameters.

Reply:
Should we add the same comments for the partial_hash_index.cpp and partial_hash_index_impl.cpp as they have a similar interface?